econml/ortho_forest.py

```diff
@@ -56,10 +56,17 @@ def _build_tree_in_parallel(Y, T, X, W,
 
 def _fit_weighted_pipeline(model_instance, X, y, sample_weight):
     if not isinstance(model_instance, Pipeline):
-        model_instance.fit(X, y, sample_weight)
+        try:
+            model_instance.fit(X, y, sample_weight=sample_weight)
+        except Exception:
```

Comment:
1.  Can we catch a more specific exception here?  I can imagine that other issues might also hit this code path, making the error message that you generate misleading.
2.  Do we even need to catch and rethrow rather than just relying on the existing error now that you're using a keyword argument?
3.  If you are going to do this, consider preserving the original exception context by using the `raise ... from ...` syntax

Reply:
Also, if we're going to emit our own error, it might be nice to suggest using a WeightedModelWrapper as a workaround.

Reply:
Great suggestion! Fixed.